stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt

```diff
@@ -760,25 +761,54 @@ internal class ChatDomainImpl internal constructor(
                 message.updatedAt != null || message.updatedLocallyAt != null -> {
                     client.updateMessage(message).execute()
                 }
-                else -> channelClient.sendMessage(message).execute()
+                else -> {
+                    val hasPendingAttachments = message.attachments.any {
```

Comment:
What do you think about adding `Message.attachmentSyncStatus` to Message? If the state is sync needed then we can start our worker, if sync is in progress then we do nothing, if attachments sync failed then we can mark the message as failed permanently.

Reply:
Makes sense to me.